elf_binary.cc

```diff
@@ -66,23 +66,33 @@ Range ELFBinary::GetRange() const {
     return range;
 }
 
-bool ELFBinary::InTLS(uintptr_t offset) const {
+bool ELFBinary::IsVaddrInTLSData(uintptr_t vaddr) const {
     if (tls_) {
-        return tls_->p_vaddr <= offset && offset < tls_->p_vaddr + tls_->p_memsz;
+        uint64_t a = reinterpret_cast<uint64_t>(vaddr);
+        uint64_t s = reinterpret_cast<uint64_t>(tls_->p_vaddr);
+        uint64_t e = reinterpret_cast<uint64_t>(tls_->p_vaddr + tls_->p_filesz);
+        bool ret = s <= a && a < e;
+        return ret;
     }
     return false;
 }
 
-bool ELFBinary::InTLSData(uintptr_t tls_offset) const {
-    if (tls_ && tls_offset < tls_->p_memsz) {
-        return tls_offset < tls_->p_filesz;
+bool ELFBinary::IsOffsetInTLSData(uintptr_t tls_offset) const {
+    uint64_t o = reinterpret_cast<uint64_t>(tls_offset);
+    uint64_t m = tls_ ? reinterpret_cast<uint64_t>(tls_->p_memsz) : 0;
+    uint64_t f = tls_ ? reinterpret_cast<uint64_t>(tls_->p_filesz) : 0;
+    if (tls_ && o < m) {
```

Comment:
[optional] `CHECK(tls_); CHECK_LT(o, m); return o < f;` is easier to read for me, but it'd be a matter of preference.